### PR TITLE
 feat: Create a task when requesting a trial - MEED-2868 - meeds-io/MIP#102

### PR DESCRIPTION
**Pull request by MayTekayaa**
This change will create a task after requesting a trial, to be treated manually then.

### CODE REVIEW

File: deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.time.ZoneOffset;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.meeds.deeds.constant.TrialStatus;
+import io.meeds.deeds.elasticsearch.model.TrialContactInformation;
+import io.meeds.deeds.storage.TrialRepository;
+
+@Component
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java]:
```suggestion
@Service
```

File: deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/model/TrialContactInformation.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import io.meeds.deeds.constant.TrialStatus;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Document(indexName = "trial_contacts")
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/model/TrialContactInformation.java]:
```suggestion
@Document(indexName = "trial_contacts", createIndex = true)
@Setting(replicas = 0, shards = 1)
```

File: deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/ElasticSearchConfig.java [hunk 3]
```diff
@@ -99,6 +100,7 @@ public ElasticsearchOperations elasticsearchOperations(ElasticsearchConverter el
       createIndex(elasticsearchTemplate, DeedMetadata.class);
       createIndex(elasticsearchTemplate, DeedTenantEvent.class);
       createIndex(elasticsearchTemplate, UserProfile.class);
+      createIndex(elasticsearchTemplate, TrialContactInformation.class);
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/ElasticSearchConfig.java]:
```suggestion
```

File: deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/ElasticSearchConfig.java [hunk 4]
```diff
@@ -38,6 +38,7 @@
 import io.meeds.deeds.elasticsearch.model.DeedSetting;
 import io.meeds.deeds.elasticsearch.model.DeedTenant;
 import io.meeds.deeds.elasticsearch.model.DeedTenantEvent;
+import io.meeds.deeds.elasticsearch.model.TrialContactInformation;
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/elasticsearch/ElasticSearchConfig.java]:
```suggestion
```

File: deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+public class TrialService {
+
+  @Autowired
+  private TrialRepository trialRepository;
+
+  @Autowired
+  private ListenerService             listenerService;
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java]: formatting
```suggestion
  @Autowired
  private TrialRepository trialRepository;

  @Autowired
  private ListenerService listenerService;
```

File: deeds-dapp-service/src/main/java/io/meeds/dapp/web/rest/TrialController.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  public TrialContactInformation saveTrial(@RequestParam(name = "firstname")
+                                           String firstname,
+                                           @RequestParam(name = "lastname")
+                                           String lastname,
+                                           @RequestParam(name = "position")
+                                           String position,
+                                           @RequestParam(name = "organization")
+                                           String organization,
+                                           @RequestParam(name = "email")
+                                           String email) {
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/dapp/web/rest/TrialController.java]: Missing code verification

```suggestion
                                           @RequestParam(name = "email")
                                           String email,
                                           @RequestParam(name = "code")
                                           long code) {
```

Comment (MayTekayaa) [reply deeds-dapp-service/src/main/java/io/meeds/dapp/web/rest/TrialController.java]: done!!

File: deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java (hunk 5 above)
Comment (MayTekayaa) [reply deeds-dapp-common/src/main/java/io/meeds/deeds/service/TrialService.java]: done!

File: deeds-dapp-common/src/main/java/io/meeds/deeds/constant/CommonConstants.java [hunk 7]
```diff
@@ -47,6 +47,10 @@ private CommonConstants() {
 
   public static final String CODE_REFRESH_HTTP_HEADER                    = "X-REFRESH";
 
+  public static final String TRIAL_CREATE_COMMAND_EVENT                  = "deed.event.trialCreate";
+
+  public static final String TRIAL_TASK_TITLE                            = "[FREE TRIAL] Request for a free trial";
```
Comment (boubaker) [inline deeds-dapp-common/src/main/java/io/meeds/deeds/constant/CommonConstants.java]: Please move this constant into Deed Provisioning Addon :
https://github.com/Meeds-io/deeds-tenant-provisioning/pull/26/files#diff-7152a4df77330d9eb4573e1d7afc728f5a73ef0c89b867402a0c728857e40573

Comment (MayTekayaa) [reply deeds-dapp-common/src/main/java/io/meeds/deeds/constant/CommonConstants.java]: done!

**Review by boubaker — APPROVED** [review]

**Pull request merged**